source_modelling/rupture_propagation.py

```diff
@@ -235,6 +211,39 @@ def select_top_spanning_trees(
     return spanning_trees
 
 
+def most_likely_spanning_tree(graph: nx.Graph) -> nx.Graph:
```

Comment:
This part is interesting, but not very clear how this works, and why you need two passes..